src/util/manifest.js

```diff
@@ -85,6 +86,15 @@ export default async function getValidatedManifest(
 
 
 export function getManifestId(manifestData: ExtensionManifest): string | void {
-  return manifestData.applications ?
-    manifestData.applications.gecko.id : undefined;
+  const manifestApps = [
+    manifestData.browser_specific_settings,
+    manifestData.applications,
+  ];
+  for (const apps of manifestApps) {
+    if (apps && apps.gecko && apps.gecko.id) {
+      return apps.gecko.id;
+    }
+  }
+
+  return undefined;
```

Comment:
Thanks! I actually like this version.

I took another look to what is being done on the Firefox side and it looks that we are doing a slightly different check here:
https://searchfox.org/mozilla-central/rev/5a4aaccb28665807a6fd49cf48367d47fbb5a19a/toolkit/mozapps/extensions/internal/XPIInstall.jsm#472-479

in particular, on the Firefox side, it looks that if browser_specific_settings.gecko is defined we only try to retrieve the id from that object and we don't look to applications.gecko.id anymore.

It may be reasonable to match that here in web-ext too, how that sounds to you?